### ISSUE TITLE
Click Events on Elements in an iFrame

### ISSUE THREAD

**Issue by spencer516** [emberjs/ember-test-helpers#330]
If you're trying to interact with content that has been loaded into an iFrame in a test, then the `click` events do not work. 

This appears to be caused by [these `instanceof` checks](https://github.com/emberjs/ember-test-helpers/blob/master/addon-test-support/%40ember/test-helpers/dom/fire-event.js#L37) that don't consider the element's realm (is that the right term?). 

If we consider this case:

```js
test('I can click on iframe elements', async function(assert) {
  await this.render(hbs`<iframe src="some-valid-source" sandbox="allow-same-origin allow-scripts"></iframe>`);
  const frame = this.element.querySelector('iframe');
  const headerElement = frame.contentDocument.querySelector('h1');

  assert.ok(headerElement instanceof Element); // fails
  assert.ok(headerElement instanceof frame.contentWindow.Element); // passes
  assert.ok(headerElement instanceof headerElement.ownerDocument.defaultView.Element); // passes
});
```

The fix should be to update those `instanceof` checks to use the element's `ownerDocument.defaultView.{Element,Document}`. 

I'm happy to put together a PR for this, but wanted to open this issue first.

**Comment by Turbo87** [comment]
wow, I didn't even know that was a thing :D

**Comment by spencer516** [comment]
I learned this very recently from a really great blog post: https://jakearchibald.com/2017/arrays-symbols-realms/

(Also, I definitely read it and thought _cool, but I'll never need to know that_. 😄)

**Comment by rwjblue** [comment]
I’d rather just check for the proper nodeType instead, I did a similar refactor recently here https://github.com/emberjs/ember-test-helpers/blob/798fa9a2b21b7466a680a0cc432aa0268ae229cc/addon-test-support/%40ember/test-helpers/dom/-get-element.js#L11

I don’t think we can do that for `window` itself, but for element and document it should work fine...

**Comment by spencer516** [comment]
Just opened https://github.com/emberjs/ember-test-helpers/pull/331 to address. Thanks!!

**Comment by rwjblue** [comment]
Fixed by #331
